src/js/components/sharedComponents/checkbox/PrimaryCheckboxType.jsx

```diff
@@ -0,0 +1,188 @@
+
+import SecondaryCheckboxType from './SecondaryCheckboxType';
+import CollapsedCheckboxType from './CollapsedCheckboxType';
+import SingleCheckboxType from './SingleCheckboxType';
+
+const propTypes = {
+    id: React.PropTypes.string,
+    name: React.PropTypes.string,
+    filters: React.PropTypes.array,
+    bulkRecipientTypeChange: React.PropTypes.func,
+    value: React.PropTypes.string,
+    filterType: React.PropTypes.string,
+    selectedCheckboxes: React.PropTypes.object
+};
+
+const defaultProps = {
+    id: '',
+    name: '',
+    filters: [],
+    value: '',
+    filterType: '',
+    selectedCheckboxes: new Set()
+};
+
+const ga = require('react-ga');
+
+export default class PrimaryCheckboxType extends React.Component {
+    static logPrimaryTypeFilterEvent(type, filter) {
+        ga.event({
+            category: 'Search Page Filter Applied',
+            action: `Selected ${filter} Type`,
+            label: type
+        });
+    }
+
+    static logDeselectFilterEvent(type, filter) {
+        ga.event({
+            category: 'Search Page Filter Applied',
+            action: `Deselected ${filter} Type Children`,
+            label: type
+        });
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showSubItems: false,
+            arrowState: 'collapsed',
+            selectedChildren: false,
+            allChildren: false
+        };
+
+        // bind functions
+        this.toggleSubItems = this.toggleSubItems.bind(this);
+        this.toggleChildren = this.toggleChildren.bind(this);
+    }
+
+    componentDidMount() {
+        this.compareFiltersToChildren();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.selectedCheckboxes.hashCode() !== this.props.selectedCheckboxes.hashCode()) {
+            this.compareFiltersToChildren();
+        }
+    }
+
+    compareFiltersToChildren() {
+        // check to see if the children are all selected or not
+        let allSelected = true;
+        let someSelected = false;
+
+        for (const code of this.props.filters) {
+            if (!this.props.selectedCheckboxes.has(code)) {
+                allSelected = false;
+            }
+            else {
+                someSelected = true;
+            }
+        }
+
+        this.setState({
+            allChildren: allSelected,
+            selectedChildren: someSelected
+        });
+    }
+
+    toggleSubItems() {
+        const newShowState = !this.state.showSubItems;
+        let newArrowState = 'collapsed';
+        if (newShowState) {
+            newArrowState = 'expanded';
+        }
+
+        this.setState({
+            showSubItems: newShowState,
+            arrowState: newArrowState
+        });
+    }
+
+    toggleChildren() {
+        let showChildren = true;
+        let arrowState = 'expanded';
+
+        if (this.state.allChildren) {
+            // all the children are selected, deselect them
+            this.props.bulkRecipientTypeChange({
```

Comment:
I thought this was supposed to be a generic component. This looks like recipient-specific code.